.github/workflows/step_coverage.yml

```diff
@@ -40,6 +43,7 @@ jobs:
       - name: Upload the coverage
         uses: codecov/codecov-action@v3
         with:
+          token: ${{ secrets.CODECOV_TOKEN }}
           flags: ${{ matrix.coverage }}
```

Comment:
Can you try adding `file: ./coverage.xml` directive to see if it fixes coverage issues? I remember having similar issue like that when using flags and fixed it by explicitly setting `-f` flag to coverage CLI.

Reply:
It's a red-herring, the same codecov command works when restarting the workflow. Codecov itself is failing there, see the top-comment. My suspicion is that there is a combination of missing `env-var` (or other method to distinguish the uploads) and race-condition + rate-limiting. Hopefully with `v4` there won't be any token needed and there would be less rate-limiting issues.